src/pull/express.ts

```diff
@@ -0,0 +1,46 @@
+import Pyroscope, { PyroscopeConfig } from '../index'
+import { Request, Response, NextFunction } from 'express'
+import debug from 'debug'
+
+const log = debug('pyroscope')
+
+async function handlerCpu(req: Request, res: Response) {
+  log('Collecting Cpu for', req.query.seconds)
+  try {
+    const p = await Pyroscope.collectCpu(Number(req.query.seconds))
+    res.send(p)
+  } catch (e) {
+    log('Error collecting cpu', e)
+    res.sendStatus(500)
+  }
+  res.end()
+}
+
+async function handlerHeap(req: Request, res: Response) {
+  log('Fetching Heap Profile')
+  try {
+    const p = await Pyroscope.collectHeap()
+    res.send(p)
+  } catch (e) {
+    log('Error collecting Heap', e)
+    res.sendStatus(500)
+  }
+  res.end()
+}
+
+export default function expressMiddleware(
+  options: PyroscopeConfig
+): (req: Request, res: Response, next: NextFunction) => void {
+  Pyroscope.init({ ...options, autoStart: false })
+  Pyroscope.startHeapCollecting()
+
+  return (req: Request, res: Response, next: NextFunction) => {
+    if (req.method === 'GET' && req.path === '/debug/pprof/profile') {
+      return handlerCpu(req, res).then(() => next())
+    }
+    if (req.method === 'GET' && req.path === '/debug/pprof/heap') {
+      return handlerHeap(req, res).then(() => next())
+    }
```

Comment:
Question: Is there a reason somebody would only want to expose, let's say, the cpu endpoint? If so, should we enable as a flag?

Reply:
Yeah, as a user I would appreciate that. Let me create an issue